### PR TITLE
Added checkPassword_1 stored procedure

### PR DESCRIPTION
**Pull request by eoger**
Part of https://github.com/mozilla/fxa-auth-db-server/issues/88

### CODE REVIEW

File: lib/db/schema/patch-010-011.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-- Add checkPassword stored procedure
+
+CREATE PROCEDURE `checkPassword_2` (
+    IN `inUid` BINARY(16),
```
Comment (chilts) [inline lib/db/schema/patch-010-011.sql]: Could this be `checkPassword_1`? I guess you still had a version 1 in your old patch, but we don't yet have on v1 in master so best to start there.

File: lib/db/mysql.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ module.exports = function (log, error) {
     // Where  : name = $1
     var DB_METADATA = 'CALL dbMetadata_1(?)'
 
-    return mysql.readFirstResult(DB_METADATA, options.patchKey)
+    return mysql.readFirstResult(DB_METADATA, [options.patchKey])
```
Comment (chilts) [inline lib/db/mysql.js]: Happy if you'd like to change all of these in a separate patch, but probably best not to bundle these changes with this patch (ditto with all the `[id]` et al below too). Hope that's ok. To be honest, I  think `param` or `[param]` works fine with the driver so am not sure if this change gives us anything anyway.

**Comment by eoger** [comment]
I divided the pull request in 2 patches like you suggested, I couldn't get the tests working without changing the param argument in read(). I also decremented the checkPassword version to 1.
Tell me what you think!

**Comment by eoger** [comment]
I also added my fxa-auth-db-server fork to npm-shrinkwrap.json so travis can test everything.

**Comment by chilts** [comment]
This looks good. Am still not sure why your stuff wasn't working without those square brackets everywhere ... but I guess this doesn't harm anything. Thanks for putting in a separate commit. Once you've put package.json back to it's former glory, I'll merge. r+

**Comment by eoger** [comment]
package.json back to normal :+1:

**Pull request merged**